File: 1751-slowest-key/1751-slowest-key.java
class Solution {
    public char slowestKey(int[] releaseTimes, String keysPressed) {
        int longest = releaseTimes[0];
        int key = (int) keysPressed.charAt(0);
        for(int i = 1; i < releaseTimes.length; i++) {
            int press = releaseTimes[i] - releaseTimes[i - 1];
            int ascii = (int) keysPressed.charAt(i);
            if(press > longest || (press == longest && ascii > key)){
                longest = press;
                key = ascii;
            }
        }
        return (char)key;
    }
}

File: 1751-slowest-key/1751-slowest-key.py
class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        longest = releaseTimes[0]
        key = ord(keysPressed[0])
        for i in range(1, len(keysPressed)):
            press = releaseTimes[i] - releaseTimes[i - 1]
            if press > longest or press == longest and ord(keysPressed[i]) > key:
                longest = press
                key = ord(keysPressed[i])
        return chr(key)
